artemis-server/src/main/java/org/apache/activemq/artemis/core/remoting/server/impl/RemotingServiceImpl.java

```diff
@@ -626,7 +626,12 @@ public void bufferReceived(final Object connectionID, final ActiveMQBuffer buffe
          ConnectionEntry conn = connections.get(connectionID);
 
          if (conn != null) {
-            conn.connection.bufferReceived(connectionID, buffer);
+            try {
+               conn.connection.bufferReceived(connectionID, buffer);
+            } catch (RuntimeException e) {
+               ActiveMQServerLogger.LOGGER.warn("Failed to decode buffer, disconnect immediately.", e);
```

Comment:
This should be done through a custom method on ActiveMQServerLogger.LOGGER rather than through the generic warn() method.